addpage.php

```diff
@@ -0,0 +1,54 @@
+<?php
+require_once('functions.php');
+$db = getDB();
+$plant_type_list = listPlantTypes(getPlantTypes($db));
+
+$error_message = '';
+$types_message = '';
+
+if (!empty($_GET)) {
+    if ($_GET['entry_add_error'] === '1') {
```

Comment:
this could be done with a function and an array of errors.
if your array has keys that match your error codes then you can just return them straight from the array. Brdley can help with this if you need